File: Problems/0009. Palindrome Number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if(x < 0)   return false;

        if(x >= 0 && x < 10)    return true;

        int tempx = x;
        long long y = 0;

        while (tempx != 0){
            int temp = tempx % 10;
            y = y * 10 + temp;
            tempx /= 10;
        }

        return (x == y);

    }  
    
};

File: Problems/0009. Palindrome Number.py
# Solution 1
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        elif x >= 0 and x < 10:
            return True
        
        
        y = x
        z = 0
        while y != 0:
            temp = y % 10
            y//= 10
            z = z * 10 + temp
        
        return x == z

# Solution 2
class Solution:
    def isPalindrome(self, x: int) -> bool:
        y = str(x)
        return y == y[::-1]

